File: leetcode/top-interview-questions/easy/array/move-zeroes.py
class Solution:
  def moveZeroes(self, nums: List[int]) -> None:
    """
    Do not return anything, modify nums in-place instead.
    """
    ans = []
    for num in nums:
      if num != 0:
        ans.append(num)
    for i in range(len(nums) - len(ans)):
      ans.append(0)
    for i in range(len(nums)):
      nums[i] = ans[i]

File: leetcode/top-interview-questions/easy/array/move-zeroes.rs
impl Solution {
  pub fn move_zeroes(nums: &mut Vec<i32>) {
      let mut z = Vec::new();
      nums.retain(|x| {
          if *x != 0 { true } else {
              z.push(0);
              false
          }
      });
      nums.append(&mut z);
  }
}